zebra/rtadv.c

```diff
@@ -632,45 +633,58 @@ static void rtadv_process_advert(uint8_t *msg, unsigned int len,
 
 	radvert = (struct nd_router_advert *)msg;
 
-	if ((radvert->nd_ra_curhoplimit && zif->rtadv.AdvCurHopLimit)
-	    && (radvert->nd_ra_curhoplimit != zif->rtadv.AdvCurHopLimit)) {
+	monotime(&now);
+	past = now;
+	past.tv_sec -= (6 * 60 * 60);
+
+	if ((radvert->nd_ra_curhoplimit && zif->rtadv.AdvCurHopLimit) &&
+	    (radvert->nd_ra_curhoplimit != zif->rtadv.AdvCurHopLimit) &&
+	    timercmp(&past, &zif->rtadv.lastadvcurhoplimit, >)) {
```

Comment:
Might be a little clearer to use `monotime_since` instead, like:

```
#define HOUR2USEC 3600000000
...
if (monotime_since(&zif->rtadv.lastadvcurhoplimit,  NULL) > (6 * HOUR2USEC)))
```

This way you only need the `now` timeval, to store it in the `lastxxx` fields where needed, instead of needing both `now` and `past`. It's simpler and eliminates what I'm pretty sure are bugs in the rest of the PR where you compare against `now` instead of `past`.

Reply:
fixed